cli/src/Api.ts

```diff
@@ -0,0 +1,135 @@
+import BN from 'bn.js';
+import { registerJoystreamTypes } from '@joystream/types';
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { formatBalance } from '@polkadot/util';
+import { Balance, Hash } from '@polkadot/types/interfaces';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { AccountBalances, CouncilInfoObj, CouncilInfoTuple, AccountSummary, createCouncilInfoObj } from './Types';
+import { DerivedFees, DerivedBalances } from '@polkadot/api-derive/types';
+import { CLIError } from '@oclif/errors';
+import ExitCodes from './ExitCodes';
+
+const API_URL = process.env.WS_URL || (
+    process.env.MAIN_TESTNET ?
+        'wss://rome-rpc-endpoint.joystream.org:9944/'
+        : 'wss://rome-staging-2.joystream.org/staging/rpc/'
+);
+const TOKEN_SYMBOL = 'JOY';
+
+// Api wrapper for handling most common api calls and allowing easy API implementation switch in the future
+
+export default class Api {
+    private _api: ApiPromise | null = null;
```

Comment:
Revisit the role of these two state variables after you have looked at the comment about in `getApi`. Their values seem to probably have some invariant over it that can be enforced statically (e.g. `_api` is null if and only if `initializing` is true, or something like that´).